### PR TITLE
WIP upgrade to angular 8

### PR DESCRIPTION
**Pull request by elizurmach**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -304,9 +304,7 @@
             "styles": [
```
Comment (asafMasa) [inline angular.json]: revert the file

File: package.json [hunk 2]
```diff
@@ -51,6 +52,7 @@
     "@ngrx/store-devtools": "^7.2.0",
     "@ngx-translate/core": "^10.0.2",
     "@turf/turf": "^5.1.6",
+    "@webcomponents/custom-elements": "^1.3.1",
```
Comment (asafMasa) [inline package.json]: undo - not relevant for this PR

File: package.json [hunk 3]
```diff
@@ -59,6 +61,7 @@
     "crypto-js": "3.1.9-1",
     "d3": "^5.9.2",
     "dltjs": "^0.1.1",
+    "document-register-element": "^1.7.2",
```
Comment (asafMasa) [inline package.json]: undo - not relevant for this PR

File: src/app/@ansyn/ansyn/ansyn.module.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { CommonModule } from '@angular/common';
 import { HttpClientModule } from '@angular/common/http';
-import { ModuleWithProviders, NgModule } from '@angular/core';
+import { Injector, ModuleWithProviders, NgModule } from '@angular/core';
+import { createCustomElement } from '@angular/elements';
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/ansyn.module.ts]: undo - not relevant for this PR

File: src/app/@ansyn/ansyn/ansyn.module.ts [hunk 5]
```diff
@@ -113,7 +120,9 @@ export class AnsynModule {
 		};
 	}
 
-	constructor(public translate: TranslateService) {
+	constructor(public translate: TranslateService, private injector: Injector) {
 		translate.setDefaultLang('default');
+		const element = createCustomElement(AnsynComponent, { injector: this.injector });
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/ansyn.module.ts]: undo - not relevant for this PR

File: src/app/@ansyn/ansyn/app-effects/effects/cases.app.effects.ts [hunk 6]
```diff
@@ -53,7 +53,7 @@ export class CasesAppEffects {
 	onDisplayOverlay$: Observable<any> = this.actions$.pipe(
 		ofType<DisplayOverlaySuccessAction>(OverlaysActionTypes.DISPLAY_OVERLAY_SUCCESS),
 		withLatestFrom(this.store$.select(mapStateSelector), this.store$.select(toolsStateSelector)),
-		map(([action, mapState, toolsState]: [DisplayOverlayAction, IMapState, IToolsState]) => {
+		map(([action, mapState, toolsState]: [DisplayOverlaySuccessAction, IMapState, IToolsState]) => {
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/app-effects/effects/cases.app.effects.ts]: why?

File: src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts [hunk 7]
```diff
@@ -135,12 +135,12 @@ export class ToolsAppEffects {
...
 	@Effect()
 	updateImageProcessingOnTools$: Observable<any> = this.activeMap$.pipe(
-		filter((map) => Boolean(map.data.overlay)),
+		filter<any>((map) => Boolean(map.data.overlay)),
 		withLatestFrom(this.store$.select(toolsStateSelector).pipe(pluck<IToolsState, ImageManualProcessArgs>('manualImageProcessingParams'))),
 		mergeMap<any, any>(([map, manualImageProcessingParams]: [ICaseMapState, ImageManualProcessArgs]) => {
 			const actions = [new EnableImageProcessing(), new SetAutoImageProcessingSuccess(map.data.isAutoImageProcessingActive)];
 			if (!isEqual(map.data.imageManualProcessArgs, manualImageProcessingParams)) {
-				actions.push(new SetManualImageProcessing(map.data && map.data.imageManualProcessArgs || this.defaultImageManualProcessArgs));
+				actions.push(new SetAutoImageProcessingSuccess(map.data !=null && map.data.imageManualProcessArgs != null || this.defaultImageManualProcessArgs != null));
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts]: why?

File: src/app/@ansyn/ansyn/modules/core/forms/file-input/file-input.component.ts [hunk 8]
```diff
@@ -16,7 +16,7 @@ import { UUID } from 'angular2-uuid';
 	]
 })
 export class FileInputComponent implements ControlValueAccessor {
-	@ViewChild('input') public input: ElementRef<any>;
+	@ViewChild('input', {static:false}) public input: ElementRef<any>;
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/core/forms/file-input/file-input.component.ts]: what "static:false' means?

File: src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts [hunk 9]
```diff
@@ -139,7 +139,7 @@ export class DisplayOverlayAction implements Action {
 }
 
 export class DisplayOverlaySuccessAction extends DisplayOverlayAction {
-	type = OverlaysActionTypes.DISPLAY_OVERLAY_SUCCESS;
+	type = OverlaysActionTypes.DISPLAY_OVERLAY;
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts]: WHY???

File: src/app/@ansyn/ansyn/modules/overlays/components/timeline/timeline.component.ts [hunk 10]
```diff
@@ -62,7 +62,7 @@ export interface IEventDropsEvent {
 
 export class TimelineComponent implements OnInit, OnDestroy {
 
-	@ViewChild('context') context: ElementRef;
+	@ViewChild('context', { static: true }) context: ElementRef;
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/overlays/components/timeline/timeline.component.ts]: what "{ static: true }" means?

File: src/polyfills.ts [hunk 11]
```diff
@@ -46,6 +46,13 @@ import 'core-js/es7/reflect';
  */
 import 'zone.js/dist/zone'; // Included with Angular CLI.
 
+/* CUSTOM ELEMENTS */
+
```
Comment (asafMasa) [inline src/polyfills.ts]: undo - it's not relevant for this PR

File: src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts (hunk 9 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts]: It is required by angular 8

File: src/app/@ansyn/ansyn/modules/core/forms/file-input/file-input.component.ts (hunk 8 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/modules/core/forms/file-input/file-input.component.ts]: It is required by angular 8

**Comment by ramtob** [comment]
There is a more up to date version of this PR, by me.

**Pull request closed without merging**